main/webapp/modules/core/scripts/project.js

```diff
@@ -591,6 +591,45 @@ Refine.getPermanentLink = function() {
   return "project?" + params.join("&");
 };
 
+(function() {
+  const systemInfo = {
+    osVersion: "",
+    availableMemory: 0,
+    totalMemory: 0,
+  };
+
+  systemInfo.byteToMB = function (bytes) {
+    return (bytes / 1024 / 1024).toFixed(2);
+  };
+
+  systemInfo.memoryWarn = function() {
+    return (this.availableMemory/this.totalMemory) * 100 < 5;
+  }
+
+  let intervalId;
+  intervalId = setInterval(function() {
+    $.ajax({
+      url: "command/core/get-system-info?",
+      type: "GET",
+      async: false,
+      data: {
+      },
+      success: function(data) {
+        systemInfo.osVersion = data["os_version"];
+        systemInfo.availableMemory = data["available_memory"];
+        systemInfo.totalMemory = data["total_memory"];
+      }
+    }).fail(function(jqXhr, textStatus, errorThrown ) {
+      alert( textStatus + ':' + errorThrown );
+      clearInterval(intervalId);
+    });
+
+    if(systemInfo.memoryWarn()) {
+      alert(`Your memory is running out. ${systemInfo.byteToMB(systemInfo.availableMemory)}MB is left.\n`)
```

Comment:
A few thoughts here:
* the message should be translatable with i18n
* A javascript alert is perhaps a bit too violent for this. Maybe a non-modal notification (like those displayed when an operation is executed) could work too?
* The message is not very actionable: what sort of action should we suggest to the user? Increase the amount of memory allocated to OpenRefine? Close some projects and let OpenRefine unload them? I am not sure we have a workable workflow for that, that's probably out of scope for this issue.

Reply:
I think "Increase the amount of memory allocated to OpenRefine" is a good fit for now

Reply:
Makes sense. Ideally we should add a link to the documentation to explain how, but that is perhaps not so easy in an "alert" dialog. Perhaps that is another reason to use something less blocking, in the DOM (see above)?

Reply:
Is this the right link? https://docs.openrefine.org/manual/running#starting-with-modifications